make/command

```diff
@@ -7,7 +7,15 @@ bin/cmdstan/%.o : src/cmdstan/%.cpp
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : CPPFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDFLAGS_MPI =
 bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : LDLIBS_MPI =
-bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : bin/%$(EXE) : bin/cmdstan/%.o
+bin/print$(EXE) bin/stansummary$(EXE) bin/diagnose$(EXE) : bin/%$(EXE) : bin/cmdstan/%.o $(BOOST)/stage/lib/libboost_program_options.a
```

Comment:
Technically libboost_program_options.a is a prerequisite for stansummary only, but we should use it everywhere since we have it. And it doesnt make a hufe difference.